src/frontend/src/lib/derived/network-tokens.derived.ts

```diff
@@ -25,13 +26,16 @@ export const combinedDerivedSortedNetworkTokens: Readable<Token[]> = derived(
 
 /**
  * All tokens matching the selected network or Chain Fusion, with the ones with non-null balance at the top of the list.
+ * It will not execute the sorting if the balance store is not yet initiated to avoid looping without data.
  */
 export const combinedDerivedSortedNetworkTokensUi: Readable<TokenUi[]> = derived(
 	[combinedDerivedSortedNetworkTokens, balancesStore, exchanges],
 	([$enabledNetworkTokens, $balances, $exchanges]) =>
-		pinTokensWithBalanceAtTop({
-			$tokens: $enabledNetworkTokens,
-			$balances,
-			$exchanges
-		})
+		nonNullish($balances)
+			? pinTokensWithBalanceAtTop({
+					$tokens: $enabledNetworkTokens,
+					$balances,
+					$exchanges
+				})
+			: ($enabledNetworkTokens as TokenUi[])
```

Comment:
Why this cast? Shouldn't we map the tokens? What if we add fields in TokenUI in the future?

Reply:
Right! Moved internally using the `mapTokenUi` functions